src/cluster.h

```diff
@@ -292,7 +320,8 @@ typedef struct {
     unsigned char myslots[CLUSTER_SLOTS/8];
     char slaveof[CLUSTER_NAMELEN];
     char myip[NET_IP_STR_LEN];    /* Sender IP, if not all zeroed. */
-    char notused1[32];  /* 32 bytes reserved for future usage. */
+    uint16_t extensions; /* Number of extensions sent along with this packet. */
+    char notused1[16];  /* 16 bytes reserved for future usage. */
```

Comment:
@madolson the math doesn't look right.  notused1 should 30 bytes as opposed to 16 because uint16_t is 2 bytes (16 bits). This will break cluster rolling upgrades.

Reply:
It'd be good to have interoperability tests with different versions.

Reply:
@PingXie do you want to prepare a PR so we can fix it before 7.0.0-rc1?

Reply:
Yep I will work on a fix next. I will also add static asserts to make sure that we catch issues like this at compile time in the future. A proper interop test suit would be even better but I feel that it is better dealt with separately.

Reply:
PR is [here](https://github.com/redis/redis/pull/10206)

Reply:
Yeah, I apparently did the math wrong. I did check interoperability on an earlier revision, but I didn't check it after making this change. I created a separate issue to add interoperability tests, https://github.com/redis/redis/issues/10214.